src/core/manager.c

```diff
@@ -1932,11 +1932,19 @@ unsigned manager_dispatch_load_queue(Manager *m) {
         return n;
 }
 
-static bool manager_unit_cache_needs_refresh(Manager *m, Unit *u) {
-        assert(m);
+bool manager_unit_file_maybe_loadable_from_cache(Unit *u) {
```

Comment:
@keszybz changed this to only take a pointer to the unit as suggested in https://github.com/systemd/systemd/pull/16197#discussion_r447743801